src/clients/go/ci.zig

```diff
@@ -15,6 +15,27 @@ pub fn tests(shell: *Shell, gpa: std.mem.Allocator) !void {
         return error.GoFmt;
     }
 
+    if (builtin.target.os.tag == .linux) {
+        log.info("golangci-lint run", .{});
+
+        const image = "golangci/golangci-lint";
+        const tag = "latest";
+        for (0..5) |attempt| {
+            if (attempt > 0) std.time.sleep(1 * std.time.ns_per_min);
+            if (shell.exec("docker image pull {image}:{tag}", .{ .image = image, .tag = tag })) {
```

Comment:
Hm, go produces statically-liked binaries, so there isn't a material reason to use docker here, on top of our usual avoidance of docker. If we are to run golangci-lint, I'd rather we find some other way to install it.  I think it's fine to punt on github actions for this? They have some action for installing it. 


Though, I am not sure if we want `golangci-lint`, as this is third-party tooling. We _should_ be running `go vet` if we aren't already, but I'd suggest maybe holding of golangci-lint until it is clear that it is useful? 

The present error, while a true one, isn't really a big issue --- worst case, it'll lead to test failing incorrectly.

Reply:
TIL about `go vet`, and no, we don't run it already!
Switching to it since it was able to catch this present issue.